public/pages/QueryInsights/QueryInsights.js

```diff
@@ -0,0 +1,171 @@
+import React, { useState, useEffect } from 'react';
+import dateMath from '@elastic/datemath';
+import { EuiSuperDatePicker, EuiInMemoryTable } from '@elastic/eui';
+
+const QueryInsights = () => {
+  const convertTime = (unixTime) => {
+    const date = new Date(unixTime);
+    const loc = date.toDateString().split(' ');
+    return loc[1] + ' ' + loc[2] + ', ' + loc[3] + ' @ ' + date.toLocaleTimeString('en-US');
+  };
+
+  const cols = [
+    {
+      field: 'timestamp',
+      name: 'Time stamp',
+      render: (timestamp) => convertTime(timestamp),
+      sortable: true,
+      truncateText: true,
+    },
+    {
+      field: 'latency',
+      name: 'Latency',
+      render: (latency) => `${latency} ms`,
+      sortable: true,
+      truncateText: true,
+    },
+    {
+      field: 'cpu',
+      name: 'CPU usage',
+      render: (cpu) => `${cpu} ns`,
+      sortable: true,
+      truncateText: true,
+    },
+    {
+      field: 'memory',
+      name: 'Memory',
+      render: (memory) => `${memory} B`,
+      sortable: true,
+      truncateText: true,
+    },
+    {
+      field: 'indices',
+      name: 'Indices',
+      render: (indices) => indices.toString(),
+      sortable: true,
+      truncateText: true,
+    },
+    {
+      field: 'search_type',
+      name: 'Search type',
+      render: (searchType) => searchType.replaceAll('_', ' '),
+      sortable: true,
+      truncateText: true,
+    },
+    {
+      field: 'node_id',
+      name: 'Coordinator node ID',
+      sortable: true,
+      truncateText: true,
+    },
+    {
+      field: 'total_shards',
+      name: 'Total shards',
+      sortable: true,
+      truncateText: true,
+    },
+  ];
+
+  const sorting = {
+    sort: {
+      field: 'timestamp',
+      direction: 'desc',
+    },
+  };
+
+  const retrievedQueries = [];
+  const [queries, setQueries] = useState(retrievedQueries);
+
+  const defaultStart = 'now-24h';
+  const [recentlyUsedRanges, setRecentlyUsedRanges] = useState([
+    { start: defaultStart, end: 'now' },
+  ]);
+  const [loading] = useState(false);
```

Comment:
This never seems to be set?

Reply:
I think I'll need it when I connect the API but for now I'll just get rid of it.